lectures/10.HPC_and_parallel_compute/scripts/analyze_sequences.py

```diff
@@ -0,0 +1,58 @@
+import argparse
+import pathlib
+
+import pandas as pd
+
+
+def analyze_sequences(sequence: str) -> tuple:
+    """
+    This function analyzes a sequence for CpG content and returns the results.
+
+    Parameters
+    ----------
+    sequence : string
+        a string of DNA sequence
+
+    Returns
+    -------
+    c_count : int
+        the number of CpG sites in the sequence
+    raw_count : int
+        the total number of C sites in the sequence
+    """
+
+    # cast the sequence to uppercase
+    sequence = sequence.upper()
+    raw_count = 0
+    c_count = 0
+    cpg_count = 0
+    for i in enumerate(sequence):
+        if i[1] == "X":
+            c_count += 1
+        raw_count += 1
+
+    return c_count, raw_count, cpg_count
+
+
+def main():
+    # set up parser for command line arguments
+    parser = argparse.ArgumentParser(
+        description="Analyze DNA sequences for CpG content"
+    )
+    parser.add_argument("--sequence", help="A DNA sequence to analyze")
+    parser.add_argument("--output_file_name", help="The file to write the results to")
+    args = parser.parse_args()
+    sequence = args.sequence
+    output_file_name = args.output_file_name
+    c_count, raw_count, cpg_count = analyze_sequences(sequence)
+    # write the results to a file
+    output_file = pathlib.Path("../results", output_file_name)
```

Comment:
this will break on someone's machine if the folder doesn't already exist, right?